proposals/0351-regex-builder.md

```diff
@@ -1098,28 +1241,100 @@ public struct Local<Output>: RegexComponent {
 
   public init<W, C0, Component: RegexComponent>(
     @RegexComponentBuilder _ component: () -> Component
-  ) where Output == (Substring, C0), Component.Output == (W, C0)
+  ) where Output == (Substring, C0), Component.RegexOutput == (W, C0)
   
   public init<W, C0, C1, Component: RegexComponent>(
     @RegexComponentBuilder _ component: () -> Component
-  ) where Output == (Substring, C0, C1), Component.Output == (W, C0, C1)
+  ) where Output == (Substring, C0, C1), Component.RegexOutput == (W, C0, C1)
   
   // ... `O(arity)` overloads
 }
 ```
 
-For example, the following regex matches string `abcc` but not `abc`.
+</details>
+
+### Composability
+
+Let's put everything together now and parse this example bank statement.
+
+```
+CREDIT    04062020    PayPal transfer    $4.99
+CREDIT    04032020    Payroll            $69.73
+DEBIT     04022020    ACH transfer       $38.25
+DEBIT     03242020    IRS tax payment    $52249.98
+```
+
+Here we have 2 types of transaction kinds, CREDIT and DEBIT, we have a date
+denoted by mmddyyyy, a description, and the amount paid.
 
 ```swift
-Regex {
-  "a"
-  Local {
+enum TransactionKind: String {
+  case credit = "CREDIT"
+  case debit = "DEBIT"
+}
+
+struct Date {
+  var month: Int
+  var day: Int
+  var year: Int
+
+  init?(mmddyyyy: String) {
+    ...
+  }
+}
+
+let statementRegex = Regex {
+  // First, lets capture the transaction kind by wrapping our ChoiceOf in a
+  // TryCapture because we want 
```

Comment:
Oops, this is my fault I forgot to finish the sentence in my document. Something like, "First, lets capture the transaction kind by wrapping our ChoiceOf in a TryCapture because our initializer can return nil on failure."